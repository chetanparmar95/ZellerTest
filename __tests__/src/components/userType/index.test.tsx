import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import UserType from '../../../../src/components/userType';
import {TouchableOpacity} from 'react-native';

describe('renders correctly', () => {
  test('Render when selected', () => {
    const wrapper = renderer.create(
      <UserType title={'Admin'} selected={true} onSelected={jest.fn()} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Render when not selected', () => {
    const wrapper = renderer.create(
      <UserType title={'Manager'} selected={false} onSelected={jest.fn()} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('onSelected calls when pressed', () => {
    const mockSelected = jest.fn();
    const wrapper = shallow(
      <UserType title={'Manager'} selected={false} onSelected={mockSelected} />,
    );
    wrapper.find(TouchableOpacity).simulate('press');
    expect(mockSelected).toHaveBeenCalledWith('Manager');
  });
});
