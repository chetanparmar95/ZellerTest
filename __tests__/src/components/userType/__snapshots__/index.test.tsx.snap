// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Render when not selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 10,
      "flex": 1,
      "flexDirection": "row",
      "marginVertical": 4,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "rgb(48,111,199)",
          "borderRadius": 12,
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
        {
          "borderColor": "rgb(234,242,250)",
        },
      ]
    }
  />
  <Text
    style={
      {
        "marginHorizontal": 16,
      }
    }
  >
    Manager
  </Text>
</View>
`;

exports[`renders correctly Render when selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgb(234,242,250)",
      "borderRadius": 10,
      "flex": 1,
      "flexDirection": "row",
      "marginVertical": 4,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "rgb(48,111,199)",
          "borderRadius": 12,
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
        {
          "borderColor": "rgb(48,111,199)",
        },
      ]
    }
  >
    <View
      style={
        {
          "backgroundColor": "rgb(48,111,199)",
          "borderRadius": 8,
          "height": 16,
          "width": 16,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "marginHorizontal": 16,
      }
    }
  >
    Admin
  </Text>
</View>
`;
